force-app/main/default/classes/Callable_API_TEST.cls

```diff
@@ -117,6 +119,68 @@ private with sharing class Callable_API_TEST {
         Test.stopTest();
     }
 
+    @IsTest
+    private static void shouldCallEnableAdvancedMappingOnAction() {
+        Callable_API callableApi = new Callable_API();
+        AdvancedMappingEnablementServiceMock mappingServiceMock = new AdvancedMappingEnablementServiceMock();
+        callableApi.advancedMappingEnablementService = stubFor(mappingServiceMock);
+
+        Test.startTest();
+            callableApi.call(ADVANCED_MAPPING_ACTION, null);
+        Test.stopTest();
+
+        System.assert(mappingServiceMock.enableAdvancedMappingCalled, 'The enable advanced mapping service method ' +
+                'should be called.');
+    }
+
+    @IsTest
+    private static void shouldNotCallEnableAdvancedMappingIfMappingIsEnabled() {
+        Callable_API callableApi = new Callable_API();
+        AdvancedMappingEnablementServiceMock mappingServiceMock = new
+                AdvancedMappingEnablementServiceMock();
+        mappingServiceMock.setEnabled();
+
+        callableApi.advancedMappingEnablementService = stubFor(mappingServiceMock);
+
+        Test.startTest();
+            callableApi.call(ADVANCED_MAPPING_ACTION, null);
+        Test.stopTest();
+
+        System.assert(!mappingServiceMock.enableAdvancedMappingCalled, 'The enable advanced mapping service method ' +
+                'should not be called.');
+    }
+
+    @IsTest
+    private static void shouldCallEnableGiftEntryOnAction() {
+        Callable_API callableApi = new Callable_API();
+        GiftEntryEnablementServiceMock giftEntryEnablementServiceMock = new GiftEntryEnablementServiceMock();
+
+        callableApi.giftEntryEnablementService = stubFor(giftEntryEnablementServiceMock);
+
+        Test.startTest();
+            callableApi.call(GIFT_ENTRY_ACTION, null);
+        Test.stopTest();
+
+        System.assert(giftEntryEnablementServiceMock.enableGiftEntryCalled, 'The enable gift entry service method ' +
+            'should be called.');
+    }
+
+    @IsTest
+    private static void shouldNotCallEnableGiftEntryIfAlreadyEnabled() {
+        GiftEntryEnablementServiceMock giftEntryEnablementServiceMock = new GiftEntryEnablementServiceMock();
+        giftEntryEnablementServiceMock.setEnabled();
+
+        Callable_API callableApi = new Callable_API();
+        callableApi.giftEntryEnablementService = stubFor(giftEntryEnablementServiceMock);
+
+        Test.startTest();
+            callableApi.call(ADVANCED_MAPPING_ACTION, null);
```

Comment:
Do you want to verify that the call throws the exception (and wouldn't this fail since an exception is thrown)?

Reply:
@force2b In these tests, we are only verifying that the method is called since the primary responsibility of the Callable_API is to call the methods. And since the method is stubbed, it's just returning a "called" `boolean` so it wouldn't fail. Then we have the underlying service test class to actually test the method execution.

Reply:
Ah. Very good pont.